public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -116,41 +110,338 @@ const Policy: FC<PolicyComponentProps> = ({
   const actualContactPoint = contactPoint ?? inheritedProperties?.receiver ?? '';
   const contactPointErrors = contactPointsState ? getContactPointErrors(actualContactPoint, contactPointsState) : [];
 
+  const childPolicies = currentRoute.routes ?? [];
+  const [showExportDrawer, toggleShowExportDrawer] = useToggle(false);
+  const matchingAlertGroups = matchingInstancesPreview?.groupsMap?.get(currentRoute.id);
+  // sum all alert instances for all groups we're handling
+  const numberOfAlertInstances = matchingAlertGroups
+    ? sumBy(matchingAlertGroups, (group) => group.alerts.length)
+    : undefined;
+
+  // simplified routing permissions
+  const [isSupportedToSeeAutogeneratedChunk, isAllowedToSeeAutogeneratedChunk] = useAlertmanagerAbility(
+    AlertmanagerAction.ViewAutogeneratedPolicyTree
+  );
+  // collapsible policies variables
+  const isThisPolicyCollapsible = useShoudPolicyBeCollapsible(currentRoute);
+  const [isBranchOpen, toggleBranchOpen] = useToggle(false);
+  const renderChildPolicies = (isThisPolicyCollapsible && isBranchOpen) || !isThisPolicyCollapsible;
+
+  const groupBy = currentRoute.group_by;
+  const muteTimings = currentRoute.mute_time_intervals ?? [];
+
+  const timingOptions: TimingOptions = {
+    group_wait: currentRoute.group_wait,
+    group_interval: currentRoute.group_interval,
+    repeat_interval: currentRoute.repeat_interval,
+  };
+
   contactPointErrors.forEach((error) => {
     errors.push(error);
   });
 
-  const hasInheritedProperties = inheritedProperties && Object.keys(inheritedProperties).length > 0;
+  const isAutogeneratedPolicyRoot = isAutoGeneratedRootAndSimplifiedEnabled(currentRoute);
 
-  const childPolicies = currentRoute.routes ?? [];
+  // build the menu actions for our policy
+  const dropdownMenuActions: JSX.Element[] = useCreateDropdownMenuActions(
+    isAutoGenerated,
+    isDefaultPolicy,
+    provisioned,
+    onEditPolicy,
+    currentRoute,
+    toggleShowExportDrawer,
+    onDeletePolicy
+  );
+
+  // check if this policy should be visible. If it's autogenerated and the user is not allowed to see autogenerated
+  // policies then we should not show it. Same if the user is not supported to see autogenerated policies.
+  const showCurrentPolicy =
```

Comment:
yes, we need to check `isSupportedToSeeAutogeneratedChunk`, as `isAutoGenerated` does not takes in account the permission.

Reply:
updated the code to use the logic for hiding instead of showing. The correct code is:
```
const hideCurrentPolicy =
  isAutoGenerated && (!isAllowedToSeeAutogeneratedChunk || !isSupportedToSeeAutogeneratedChunk);
```
as we have to take in account also the `isSupportedToSeeAutogeneratedChunk` permission.